ballerina/error.bal

```diff
@@ -0,0 +1,18 @@
+// Copyright (c) 2024 WSO2 LLC. (http://www.wso2.com).
+//
+// WSO2 LLC. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+# Represents any error related to Ballerina Schema Registry module
+public type Error distinct error;
```

Comment:
Have we considered the possibility of adding additional details into the `Error` object as `details` [1] ? I think it would be quite useful to the developers to have more contextual information about application errors. 

[1] - [https://ballerina.io/learn/by-example/error-detail/](https://ballerina.io/learn/by-example/error-detail/)